Meson build failure in macOS Catalina (appleframeworks not found, tried framework) in p11-kit 0.25.1+

I've been having compilation failures since [p11-kit 0.25.1/0.25.2](https://github.com/p11-glue/p11-kit/commit/daabb9f2f6e2443e4ce3155ac75341c924175b50?diff=split) in macOS Catalina (with xcode-select installed). It was working fine with 0.25.0. Should this be reported in Meson or perhaps I am missing something on my end?

```bash
==> meson setup build -Dsystem_config=/usr/local/etc -Dmodule_config=/usr/local/etc/pkcs11/modules -Dtrust_paths=/usr/local/etc/ca-certificates/cert.pem" -Dsystemd=disabled
Last 15 lines from /Users/username/Library/Logs/Homebrew/p11-kit/01.meson:
Build type: native build
Project name: p11-kit
Project version: 0.25.2
C compiler for the host machine: clang (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C linker for the host machine: clang ld64 609.8
Host machine cpu family: x86_64
Host machine cpu: x86_64
Has header "libintl.h" : YES 
Library intl found: YES
Checking for function "dgettext" with dependency -lintl: YES 
Run-time dependency appleframeworks found: NO (tried framework)

meson.build:70:26: ERROR: Dependency "appleframeworks" not found, tried framework
```

Side note: it is used for dependencies in QEMU -> gnutls -> p11-kit. Thanks!

Quick test with check_appleframeworks.c and meson.build:

```c
#include <stdio.h>

int main() {
    printf("Apple Frameworks Test\n");
    return 0;
}
```

```meson
project('check_appleframeworks', 'c')
src = 'check_appleframeworks.c'
appleframeworks = dependency('appleframeworks', modules: 'foundation')

if appleframeworks.found()
  message('appleframeworks with "foundation" module found')
else
  message('appleframeworks with "foundation" module not found')
endif

executable('check_appleframeworks', src, dependencies: appleframeworks)
```

Output:
```bash
username appleframeworks_test % meson build
The Meson build system
Version: 1.2.3
Source dir: /Users/username/Development/appleframeworks_test
Build dir: /Users/username/Development/appleframeworks_test/build
Build type: native build
Project name: check_appleframeworks
Project version: undefined
C compiler for the host machine: cc (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C linker for the host machine: cc ld64 609.8
Host machine cpu family: x86_64
Host machine cpu: x86_64
Run-time dependency appleframeworks found: NO (tried framework)

meson.build:7:18: ERROR: Dependency "appleframeworks" not found, tried framework

A full log can be found at /Users/username/Development/appleframeworks_test/build/meson-logs/meson-log.txt
WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.
```

Error logs (meson-log.txt):
```
Build started at 2023-11-05T06:35:55.255921
Main binary: /usr/local/opt/python@3.12/bin/python3.12
Build Options: 
Python system: Darwin
The Meson build system
Version: 1.2.3
Source dir: /Users/username/Development/appleframeworks_test
Build dir: /Users/username/Development/appleframeworks_test/build
Build type: native build
Project name: check_appleframeworks
Project version: undefined
-----------
Detecting compiler via: `cc --version` -> 0
stdout:
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
-----------
Running command: cc -E -dM -
-----
-----------
Detecting linker via: `cc -Wl,--version` -> 1
stderr:
ld: unknown option: --version
clang: error: linker command failed with exit code 1 (use -v to see invocation)
-----------
-----------
Detecting Apple linker via: `cc -Wl,-v` -> 1
stderr:
@(#)PROGRAM:ld  PROJECT:ld64-609.8
BUILD 15:07:46 Dec 18 2020
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em
Library search paths:
	/usr/local/lib
	/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib
Framework search paths:
	/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
-----------
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /Users/username/Development/appleframeworks_test/build/meson-private/sanitycheckc.exe
C compiler for the host machine: cc (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C linker for the host machine: cc ld64 609.8
-----------
Detecting linker via: `ar --version` -> 1
stderr:
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
-----------
-----------
Detecting compiler via: `cc --version` -> 0
stdout:
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
-----------
Running command: cc -E -dM -
-----
-----------
Detecting linker via: `cc -Wl,--version` -> 1
stderr:
ld: unknown option: --version
clang: error: linker command failed with exit code 1 (use -v to see invocation)
-----------
-----------
Detecting Apple linker via: `cc -Wl,-v` -> 1
stderr:
@(#)PROGRAM:ld  PROJECT:ld64-609.8
BUILD 15:07:46 Dec 18 2020
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em
Library search paths:
	/usr/local/lib
	/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib
Framework search paths:
	/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
-----------
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /Users/username/Development/appleframeworks_test/build/meson-private/sanitycheckc.exe
C compiler for the build machine: cc (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C linker for the build machine: cc ld64 609.8
-----------
Detecting linker via: `ar --version` -> 1
stderr:
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
-----------
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Running compile:
Working directory:  /Users/username/Development/appleframeworks_test/build/meson-private/tmpgxg7gflj
Code:
 int main(void) { return 0; }
-----------
Command line: `cc /Users/username/Development/appleframeworks_test/build/meson-private/tmpgxg7gflj/testfile.c -o /Users/username/Development/appleframeworks_test/build/meson-private/tmpgxg7gflj/output.exe -O0 -Werror=implicit-function-declaration -framework foundation` -> 1
stderr:
ld: framework not found foundation
clang: error: linker command failed with exit code 1 (use -v to see invocation)
-----------
Run-time dependency appleframeworks found: NO (tried framework)

meson.build:7:18: ERROR: Dependency "appleframeworks" not found, tried framework
```

More information (MacOS Catalina 10.15.7):
```bash
username % xcrun --show-sdk-path 
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
username % xcode-select --install
xcode-select: error: command line tools are already installed, use "Software Update" to install updates
```

There are more issues but not sure if they are related:
[Support for appleframeworks with GCC and Intel](https://github.com/mesonbuild/meson/commit/864f49e9d03b43392b30ac207b28d6aa89281ff3)
[Cross-compilation for iOS fails after upgrade to Big Sur (macOS 11)](https://github.com/mesonbuild/meson/issues/8359)

Thanks :)

--- a/meson.build
+++ b/meson.build
@@ -67,7 +67,7 @@ if get_option('nls') and cc.has_header('libintl.h')
     if cc.has_function('dgettext', dependencies : libintl)
       libintl_deps += libintl
       if ['darwin', 'ios'].contains(host_system)
-        appleframeworks = dependency('appleframeworks', modules : 'foundation')
+        appleframeworks = dependency('appleframeworks', modules : 'CoreFoundation')
         if appleframeworks.found()
           libintl_deps += appleframeworks
         endif

